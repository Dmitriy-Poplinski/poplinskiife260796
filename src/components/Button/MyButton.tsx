import React from 'react';

import classes from './MyButton.module.css';

interface MyButtonProps {
    shuffleImageArray: () => void
}

export function MyButton({ shuffleImageArray }: MyButtonProps) {
  return (
    <button className={classes.button} onClick={() => shuffleImageArray()} data-testid="button">
        Click me
    </button>
  );
}
