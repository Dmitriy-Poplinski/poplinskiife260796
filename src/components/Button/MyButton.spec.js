import { render, screen } from '@testing-library/react';

import { MyButton } from './MyButton';

describe('MyButton:', () => {
    test('MyButton snapshot', () => {
        render(<MyButton onClick={() => {}}/>);

        expect(screen.getByTestId('button')).toMatchSnapshot();
    });
});
