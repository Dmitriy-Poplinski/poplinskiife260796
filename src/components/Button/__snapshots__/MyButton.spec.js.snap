// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyButton: MyButton snapshot 1`] = `
<button
  class="button"
  data-testid="button"
>
  Click me
</button>
`;
